testvectors/vector.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testvectors
+
+import (
+	"testing"
+)
+
+// Vector describes an individual test case. Test cases are exported for ease
+// in applying them to various implementations.
+type Vector interface {
+	Name() string
+	Valid() bool
+	// Run the Vector on cur with schema to return the resulting object and
+	// failing the test on unexpected conflicts
+	Run(t *testing.T, cur YAMLObject, schema SchemaDefinition) YAMLObject
+}
+
+// TestVector is an implementation of Vector which just returns the defined ReturnObject
```

Comment:
It's unclear what all of these things are going to be used for?

Reply:
Those are example/test implementations of the Vector (now Action) interface being used in [example.go](testvectors/example.go) to show how it works.